### ISSUE TITLE
Refactor `waitFor*` actions with `resolveSelect` 

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#9008]
## Feature Description

This issue is a reboot of #1765 which was intended for the same purpose but was more about introducing new infrastructure ("`asyncSelect`") which is now available in our version of wp data as `resolveSelect` as well as a utility action in #8992 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All use of `waitFor*` actions should be refactored with respective calls to `resolveSelect`
* All `waitFor*` actions should no longer be used and should be removed

## Implementation Brief

- [ ] Use of the Tag Manager store's [`waitForContainers()`](https://github.com/google/site-kit-wp/blob/fd2b92acec9f393b2719410f7bbdd49d3837d5bf/assets/js/modules/tagmanager/datastore/containers.js#L230-L243) action should be replaced as follows:
  - Where `waitForContainers()` being used prior to a call to `getContainers()`, it should be replaced by the use of `resolveSelect()` to resolve the `getContainers()` selector.
  - A resolver for `getWebContainers()` should be added which uses `resolveSelect()` to resolve the `getContainers()` selector.
  - Where `waitForContainers()` is being used prior to a call to `getWebContainers()`, it should be replaced by the use of `resolveSelect()` to resolve the `getWebContainers()` selector.
- [ ] Use of the site store's [`waitForHTMLForURL()`](https://github.com/google/site-kit-wp/blob/fd2b92acec9f393b2719410f7bbdd49d3837d5bf/assets/js/googlesitekit/datastore/site/html.js#L142-L147) action should be replaced by the use of `resolveSelect()` to resolve the `getHTMLForURL()` selector.
- [ ] Use of the Analytics store's [`waitForPropertySummaries()`](https://github.com/google/site-kit-wp/blob/fd2b92acec9f393b2719410f7bbdd49d3837d5bf/assets/js/modules/analytics-4/datastore/properties.js#L524-L529) action should be replaced as follows:
  - A resolver for `getPropertySummaries()` should be added which uses `resolveSelect()` to resolve the `getAccountSummaries()` selector.
  - Use of `waitForPropertySummaries()` should be replaced by the use of `resolveSelect()` to resolve the `getPropertySummaries()` selector.
- [ ] Use of the Analytics store's [`waitForWebDataStreams()`](https://github.com/google/site-kit-wp/blob/fd2b92acec9f393b2719410f7bbdd49d3837d5bf/assets/js/modules/analytics-4/datastore/webdatastreams.js#L210-L215) action should be replaced as follows:
  - A resolver for `doesWebDataStreamExist()` should be added which uses `resolveSelect()` to resolve the `getWebDataStreams()` selector.
  - A resolver for `getMatchingWebDataStreamByPropertyID()` should be added which also uses `resolveSelect()` to resolve the `getWebDataStreams()` selector.
  - Where `waitForWebDataStreams()` is being used prior to a call to `doesWebDataStreamExist()`, it should be replaced by the use of `resolveSelect()` to resolve the `doesWebDataStreamExist()` selector.
  - Where `waitForWebDataStreams()` is being used prior to a call to `getMatchingWebDataStreamByPropertyID()`, it should be replaced by the use of `resolveSelect()` to resolve the `getMatchingWebDataStreamByPropertyID()` selector.
- [ ] The above `wait`-prefixed actions, and their associated controls should be removed from the codebase.
- [ ] The [`waitForExistingTag()`](https://github.com/google/site-kit-wp/blob/fd2b92acec9f393b2719410f7bbdd49d3837d5bf/assets/js/googlesitekit/data/create-existing-tag-store.js#L92-L97) action and associated control should be removed from `createExistingTagStore()` as it's not in use.

## Test Coverage

* Remove tests for the above actions.
* Add tests to cover the newly added resolvers.
* Fix any failing tests.

## QA Brief

* Give Site Kit a smoke test, focusing particularly on the following aspects/areas:
  * Behaviour rather than presentation (no styling has been modified in this update so there's no need to go looking for pixel level changes etc).
  * Analytics and Tag Manager functionality, including "existing tag" related behaviour.
  * The AdSense Ad Blocking Recovery widget.

## Changelog entry

* N/A

**Comment by aaemnnosttv** [comment]
@techanvil I think we could introduce a few resolvers here to make the transition a bit more ergonomic where a selector relies on another selector with its resolver to get its value. Otherwise, we'll end up using `resolveSelect` in the same way which is less explicit, rather than awaiting the value we are selecting, and I'd like to avoid introducing more instances of `yield commonActions.await( ... )` where possible 😄 

E.g. 
```
getWebContainers -> getContainers -> (resolver)
```

We can update the main selector to have a resolver that works the same way but allows for using resolve select directly where we want to.

```js
resolvers.getWebContainers = function* ( accountID ) {
    yield Data.controls.resolveSelect( MODULES_TAGMANAGER, 'getContainers', accountID );
}
```
This would keep things clean and also keep the underlying resolver responsible for the actual fetch/state change with out risking duplication. 

WDYT?

**Comment by eugene-manuilov** [comment]
Do we also want to rework `waitForModules` selector as well? Seems like it's something that can be reworked with resolveSelect, isn't it?

**Comment by techanvil** [comment]
Thanks @aaemnnosttv, I think that's a fair idea. Given the fairly low number of uses of these `wait*` actions, taking your proposed approach will actually result in bit more code being added here, but it will be more ergonomic in use as you've pointed out. I've amended the IB to take this approach.

@eugene-manuilov, `waitForModules()` is already implemented using `resolveSelect()` so it doesn't look like we need to do anything here for that one:

https://github.com/google/site-kit-wp/blob/9b91d9c3d74ff327eeef3a170f29730622dbb5d8/assets/js/googlesitekit/modules/datastore/modules.js#L675-L679

**Comment by eugene-manuilov** [comment]
Yeah, i meant to refactor the usage of the `waitForModules` action.

**Comment by techanvil** [comment]
Thanks @aaemnnosttv @eugene-manuilov, however, `waitForModules()` is not an action - it's a reusable function that's used as the resolver for `getModule()`, `waitForModules()` and `isModuleConnected()`. I think  we can leave it as it is.

https://github.com/google/site-kit-wp/blob/f93a032c9ec9cb58200594e676936f9a854da6c0/assets/js/googlesitekit/modules/datastore/modules.js#L677-L683

https://github.com/google/site-kit-wp/blob/f93a032c9ec9cb58200594e676936f9a854da6c0/assets/js/googlesitekit/modules/datastore/modules.js#L802-L806

**Comment by aaemnnosttv** [comment]
@techanvil we can keep the reusable function for the resolvers, I wasn't suggesting we remove that, we just don't need the `waitForModules` action as we want to encourage using `resolveSelect` on selectors rather than leaking implementation details about what they really depend on outside of their resolvers.

**Comment by techanvil** [comment]
@aaemnnosttv, am I missing something, as I don't see a `waitForModules()` action?

**Comment by aaemnnosttv** [comment]
Ah, my mistake. I misread the snippet you shared. I think it was an action in the past but no matter. Thanks @techanvil !

IB ✅

**Comment by kelvinballoo** [comment]
### QA Update ✅ 

Verified as follows:

- Behaviour of plugin is as expected. Set up plugin with all modules and no issues were encountered.
- Analytics and Tag Manager modules are functioning appropriately.
- Analytics and Tag manager "existing tag" behaviour is as expected. Tested with 'continue with analytics' and 'continue without analytics' and no issues were encountered. 
   <Details>
   
   https://github.com/user-attachments/assets/33d5b3f5-c6c3-4753-9f8d-38c01fa33659
   
   
   https://github.com/user-attachments/assets/3a3608e4-b6f1-4f37-b020-508aaddbd67b


   </Details>

- The AdSense Ad Blocking Recovery widget is working as expected.
   - Setup from banner and from settings are working
   - Dismissal is working
   - Toggle in adsense module is working as expected
   -  Scripts are inserted accordingly in page source code based on the toggle saved
      <Details>
   
      https://github.com/user-attachments/assets/9703d186-4d82-4d51-9c49-51ebee30401d

      https://github.com/user-attachments/assets/0c62f188-7feb-4a2d-8c36-23774ed49b28

      https://github.com/user-attachments/assets/7c3f4f73-1f02-434e-86d9-02e82f86e59b

      Without protection code:
      <img width="1098" alt="Image" src="https://github.com/user-attachments/assets/dff39e74-0957-4136-88fd-d83df977f32f" />
   

      With protection:
      <img width="1091" alt="Image" src="https://github.com/user-attachments/assets/9d84affc-a131-4172-a1bd-c4871a638b2c" />

      </Details>

**Comment by tofumatt** [comment]
(Moving directly to Approval as this needs no further QA.)
